packages/web-app-pdf-viewer/src/App.vue

```diff
@@ -51,7 +51,13 @@ export default defineComponent({
     async loadPdf(fileContext) {
       try {
         this.loading = true
-        this.resource = await this.getFileInfo(fileContext)
+        const resource = await this.getFileInfo(fileContext)
+
+        if (resource.mimeType !== 'application/pdf') {
+          return
```

Comment:
note that return in try...catch...finally... block will call the finally block and sets loading to false